troubadix/standalone_plugins/no_solution.py

```diff
@@ -164,51 +164,70 @@ def extract_tags(content: str) -> Optional[Tuple[datetime, datetime, str]]:
 
     oid = oid_match.group("value")
 
-    return solution_date, creation_date, oid
+    return oid, creation_date, solution_date
 
 
-def check_no_solutions(
-    files: Iterable[Path], milestones: List[int], snooze_duration: int
-) -> List[Tuple[int, List[Tuple[Path, str, datetime, datetime]]]]:
-    summary = defaultdict(list)
+def get_no_solution_vts(
+    files: Iterable[Path],
+) -> Iterable[Tuple[Path, str, datetime, datetime]]:
+    file_contents = (
+        (file, file.read_text(encoding=CURRENT_ENCODING)) for file in files
+    )
+    return (
+        (file,) + extract_tags(content)
```

Comment:
```suggestion
        (file, *extract_tags(content))
```
This should also work and is a more readable / less confusing IMHO